osu.Game/Rulesets/UI/DrawableRuleset.cs

```diff
@@ -230,7 +230,9 @@ private void loadObjects(CancellationToken cancellationToken)
 
         public override void RequestResume(Action continueResume)
         {
-            if (ResumeOverlay != null && (Cursor == null || (Cursor.LastFrameState == Visibility.Visible && Contains(Cursor.ActiveCursor.ScreenSpaceDrawQuad.Centre))))
+            if (ResumeOverlay != null
+                && (Cursor == null || (Cursor.LastFrameState == Visibility.Visible && Contains(Cursor.ActiveCursor.ScreenSpaceDrawQuad.Centre)))
+                && Mods.All(mod => !mod.HidesResumeOverlay))
```

Comment:
I would explore the possibility of doing something so that `DrawableRuleset` can preview the active mods at the point of creating the resume overlay, and *not* create it if it determines that the active mods are incompatible with resume. It would remove the requirement of jamming even more ugly logic in here.

Note: I'm not saying to *do this* explicitly, I'm saying to *explore* it. As in: consider it, maybe try how it looks code-wise, and then push *if* the direction ends up making sense.

Reply:
Not really possible with the approach I took because the `ResumeOverlay` is set in `ApplyToDrawableRuleset` after it's already been created. But the excess ugly logic from before is gone now anyway.